third_party/nvfuser/csrc/ir_nodes.cpp

```diff
@@ -2637,37 +2637,27 @@ void TensorDomain::swizzle(
 
 std::vector<IterDomain*> TensorDomain::noReductions(
     const std::vector<IterDomain*>& td) {
-  size_t size_out = 0;
-  for (auto id : td) {
-    if (!id->isReduction() && !id->isStride()) {
-      size_out++;
-    }
-  }
-  std::vector<IterDomain*> noReductionDomain(size_out);
-
-  int it = 0;
-  for (auto id : td) {
-    if (!id->isReduction() && !id->isStride()) {
-      noReductionDomain[it++] = id;
-    }
-  }
-
+  std::vector<IterDomain*> noReductionDomain;
```

Comment:
Generally I like the copy_if version, but wouldn't the previous version be a little more efficient as it would allocate the backing buffer of the right size? I.e., `std::vector<IterDomain*> noReductionDomain(size_out);`

 Not sure how much it would matter, especially since the number of domains should be pretty small.

I know @zasdfgbnm often follows similar code patterns. He might have some feedback.